### PR TITLE
Add support for VS2022 and ARM64 compilation of packages using emsdk

### PR DESCRIPTION
**Pull request by Blackhex**
This is actually enough to compile binaryen using emsdk.

### CODE REVIEW

File: emsdk.py [hunk 1]
```diff
@@ -1133,21 +1182,11 @@ def build_fastcomp(tool):
...
-    # with a -A arch parameter.
-    args += ['-A', 'x64' if tool.bitness == 64 else 'x86']
-  elif 'Visual Studio' in CMAKE_GENERATOR and tool.bitness == 64:
-    cmake_generator += ' Win64'
-
   build_dir = llvm_build_dir(tool)
   build_root = os.path.join(fastcomp_root, build_dir)
 
   build_type = decide_cmake_build_type(tool)
+  cmake_generator, args = decide_cmake_generator(tool)
```
Comment (eukarpov) [inline emsdk.py:1190]: Why is it needed and how it was working without this change with previous VS versions?

File: emsdk.py [hunk 2]
```diff
@@ -1217,7 +1257,8 @@ def build_llvm(tool):
     targets_to_build = 'WebAssembly;AArch64'
   else:
     targets_to_build = 'WebAssembly'
-  args = ['-DLLVM_TARGETS_TO_BUILD=' + targets_to_build,
+
+  args += ['-DLLVM_TARGETS_TO_BUILD=' + targets_to_build,
```
Comment (eukarpov) [inline emsdk.py:1263]: why it was changed to += ?

File: emsdk.py [hunk 3]
```diff
@@ -1320,27 +1340,17 @@ def build_ccache(tool):
...
-
+  args += ['-DZSTD_FROM_INTERNET=ON']
   cmakelists_dir = os.path.join(src_root)
   success = cmake_configure(cmake_generator, build_root, cmakelists_dir, build_type, args)
   if not success:
     return False
 
   # Make
-  success = make_build(build_root, build_type, 'x64' if tool.bitness == 64 else 'Win32')
+  success = make_build(build_root, build_type, get_build_target_platform(tool))
```
Comment (eukarpov) [inline emsdk.py]: get_build_target_platform is used multiple times. Could it be replaced by variable?

File: emsdk.py (hunk 1 above)
Comment (Blackhex) [reply emsdk.py:1190]: The repeating code was extracted to a function.

File: emsdk.py (hunk 2 above)
Comment (Blackhex) [reply emsdk.py:1263]: The args are initialized by `decide_cmake_generator(tool)` and here the list is extended now.

File: emsdk.py (hunk 3 above)
Comment (Blackhex) [reply emsdk.py]: I'll move the call of `get_build_target_platform(tool)` to a  separate line to avoid nested function calls.

**Review by eukarpov — APPROVED** [review]

**Pull request merged**